app/models/scheduled_mailing.rb

```diff
@@ -1,34 +1,59 @@
 class ScheduledMailing < ApplicationRecord
+  # records an email for immediate or later delivery. Use .send_now! or .send_later! methods to schedule mails.
+
   serialize :arguments, Array
 
-  belongs_to :legal_aid_application
+  VALID_STATUSES = [
+    'waiting', # record has been written, but mail not yet sent by ScheduledMailingsDeliveryJob
+    'processing', # ScheduledMailingsDeliveryJob is in the process of sending the mail
+    'created', # The mail has been sent to GOVUK Notify, but the status has not yet been queries
+    'sending', # GOVUK Notify is still trying to send the mail
+    'delivered', # GOVUK Notify has confirmed that the mail has been delivered
+    'technical-failure', # GOVUK Notify has reported that the mail cannot  be sent due to a technical failure
+    'permanent-failure', # GOVUK Notify has reported a permanent failure (most likely, non-existent email address)
+    'temporary-failure', # GOVUK Notify has reported a temporary failure
+    'cancelled' # Scheduled mail cancelled because it is no longer needed to be sent.
+  ].freeze
 
-  scope :not_sent, -> { where(sent_at: nil) }
-  scope :not_cancelled, -> { where(cancelled_at: nil) }
-  scope :past_due, -> { where('scheduled_at < ?', Time.current) }
+  MONITORED_STATUSES = %w[processing created sending].freeze
 
-  def self.due_now
-    not_sent.not_cancelled.past_due
-  end
+  FAILURE_STATUSES = %w[technical-failure permanent-failure temporary-failure].freeze
```

Comment:
Do all these statuses have to be strings or could we make them all syms and replace hyphens with underscores so then they work?

Reply:
Oh I see, it's because the statuses all go into the DB and removes the issue of type conversion. Righto.